src/settings.rs

```diff
@@ -65,3 +66,223 @@ where
 {
     try_read_file_config::<T, ConfigError>(file, env_prefix).expect("unable to read config")
 }
+
+#[derive(Debug, Error)]
+pub enum SettingsError {
+    #[error("Config error: {0}")]
+    Config(#[from] config::ConfigError),
+    #[error("IO error: {0}")]
+    Io(#[from] std::io::Error),
+    #[error("bad JSON: {0}")]
+    Json(#[from] serde_json::Error),
+    #[error("bad application secret")]
+    BadSecret,
+}
+
+/// Macro for simple initialization of Settings structures.
+/// The struct inside macro define as a common way, but with little improvement. You should to type
+/// default value after the type with separator `=>` for example `pub field_name: TypeName => <default_value>`
+///
+/// # Example:
+/// ```
+/// settings_impl! {
+///     #[derive(Debug, Deserialize, PartialEq, Eq)]
+///     pub struct ExampleSettings {
+///         #[serde(default = "ExampleSettings::default_field_1")]
+///         pub some_u8_field: u8 => 1,
+///
+///         #[serde(default = "ExampleSettings::default_field_2")]
+///         pub some_string_field: String => "hello I'm example settings".into(),
+///
+///         #[serde(default = "ExampleSettings::default_logger")]
+///         pub logger: LoggerSettings => LoggerSettings::default()
+///     }
+/// }
+///
+/// fn main() {
+///     let example_settings = ExampleSettings::new();
+///
+///     assert_eq!(example_settings.some_u8_field, 1);
+///     assert_eq!(&example_settings.some_string_field, "hello I'm example settings");
+///     assert_eq!(example_settings.logger, LoggerSettings::default());
+/// }
+/// ```
+#[macro_export]
+macro_rules! settings_impl {
+    {$(
+        $( #[ $attr:meta ] )*
+        $vis:vis struct $name:ident { $(
+            $( #[ $childm:meta ] )*
+            $vis_f:vis $field:ident: $type:ident => $def:expr
```

Comment:
Idea: allow user to drop `=> $def:expr`, if he does so, expand it to `$field: <$type as Default>::default()`